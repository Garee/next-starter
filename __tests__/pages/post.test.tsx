import renderer from "react-test-renderer";
import Post from "../../pages/posts/[id]";

describe("postPage", () => {
  it("renders correctly", () => {
    expect.assertions(1);
    const postData = {
      id: "1",
      date: "2021-01-01",
      title: "Post #1",
      content: "Sample content.",
    };
    const post = <Post postData={postData} />;
    const tree = renderer.create(post).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
