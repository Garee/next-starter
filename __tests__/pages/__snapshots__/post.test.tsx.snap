// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post page renders correctly 1`] = `
<div
  className="container"
>
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTA4IiBoZWlnaHQ9IjEwOCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          alt="Gary Blackwood"
          className="borderCircle"
          data-nimg="intrinsic"
          decoding="async"
          src="/_next/image?url=%2Fimages%2Favatar.jpg&w=256&q=75"
          srcSet="/_next/image?url=%2Fimages%2Favatar.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Favatar.jpg&w=256&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="Gary Blackwood"
            className="borderCircle"
            data-nimg="intrinsic"
            decoding="async"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Favatar.jpg&w=256&q=75"
            srcSet="/_next/image?url=%2Fimages%2Favatar.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Favatar.jpg&w=256&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </div>
    </a>
    <h2
      className="headingLg"
    >
      <a
        className="colorInherit"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Gary Blackwood
      </a>
    </h2>
  </header>
  <main>
    <h1
      className="headingXl"
    >
      Post #1
    </h1>
    <small
      className="lightText"
    >
      <time
        dateTime="2021-01-01"
      >
        January 1, 2021
      </time>
    </small>
    <br />
    <small>
      ⭐⭐⭐
    </small>
    <br />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "Sample content.",
        }
      }
    />
  </main>
  <div
    className="backToHome"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      ← Back to home
    </a>
  </div>
</div>
`;
